src/main/java/com/xebialabs/restito/server/StubServer.java

```diff
@@ -43,9 +51,47 @@ public class StubServer {
      */
     public boolean secured;
 
+    /**
+     * Whether the server requires client authentication in HTTPS mode.
+     */
+    private boolean clientAuth;
+    private byte[] clientAuthTrustStore;
+    private String clientAuthTrustStorePass;
+
     private Logger log = LoggerFactory.getLogger(StubServer.class);
 
 
+    /**
+     * Get the trust store for server's certificate.
+     * @return trust store
+     */
+    public static KeyStore getTrustStore() {
```

Comment:
No, it defects the purpose of "exposing trust store for client". For the library user, they should have no knowledge of where the actual server's key store is (`keystore_server_test`). That's why this method was added at the first place. Then, `UsingHttpsTest` demonstrates how one should use it.

Reply:
@rafan, thanks for your reply.

You probably mean `keystore_server_cert`. But ok, I see your point. It's nice, indeed, if the users don't have to know where those key stores are located. The main two points I would like to improve then:

1. If we don't want to expose the server's Keystore locations, let's not do it. That means that `shouldFailSendingNonAllowedCertificate` should not use `keystore_server` but something else.
2. `StubServer` class grows heavy with TLS-related code. Since the methods are static, I propose to move them out into the class `StubServerTls` with the utility methods.

Reply:
Please check 65bb50977aa19691f54513268b7ebfd5387aab1b

Reply:
https://github.com/mkotsur/restito/commit/65bb50977aa19691f54513268b7ebfd5387aab1b looks good to me. 

Yes, I meant `keystore_server_cert`. As for `shouldFailSendingNonAllowedCertificate`, I will generate another key/cert instead.

Reply:
Check https://github.com/mkotsur/restito/pull/84/commits/a3cbfc5c127bccf1fbb0090478d211daedd63300.